### ISSUE TITLE
New pages created during development require dev server restart

### ISSUE THREAD

**Issue by josephkmh** [doczjs/docz#1450]
## Bug Report

When in development mode, new pages that are created after the development server is running are not accessible (return a Gatsby 404). The pages are accessible only after restarting the dev server.

**To Reproduce**

1. Install docz, e.g. the [Basic Docz example](https://github.com/doczjs/docz/tree/master/examples/basic)
2. Start the development server with `npm run dev`
3. Add a new page, for example: `src/test.mdx`
4. Add frontmatter properties `name` and `route`
5. See the page hot-reloaded in the sidebar menu :tada:
6. Click on the page and get a 404 :cry:

**Expected behavior**

I would expect the page to be accessible without needing to restart the dev server. Constant restarting of the dev server kind of defeats the purpose of hot-reloading. Since I am trying to quickly put together some documentation, this is a significant slowdown.

**Environment**

- docz version: 2.3.1
- OS: Ubuntu 18.04
- Node/npm version: Node 12.16.1/npm 6.14.4

**Comment by edmundojr** [comment]
Having the same issue. Bummer having to restart the dev server with `docz dev` every time you add a page.

**Comment by SimonKaluza** [comment]
Oh I think my issue here https://github.com/doczjs/docz/issues/1492 is a duplicate of this one -- I will close my issue in favor of this

**Comment by stephane-klein** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

I have also this bug.

**Comment by joshmobley** [comment]
Seems a rename or deletion of a file also blows up the dev server. This should be reopened.
